Code/Source/RobotControl/RobotControl.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+#include <memory>
+#include <rclcpp/rclcpp.hpp>
+
+#include "ROS2/ROS2Bus.h"
+#include "Utilities/ROS2Names.h"
+#include "RobotControl/ControlConfiguration.h"
+#include "RobotControl/RobotConfiguration.h"
+#include "Frame/ROS2FrameComponent.h"
+
+namespace ROS2
+{
+    // Component extension enabling polymorphic use of generics
+    // TODO - naming (this is a component activation interface capturing component and entity state)
+    class IRobotControl
+    {
+    public:
+        virtual void Activate(const AZ::Entity* entity) = 0;
+        virtual void Deactivate() = 0;
+        virtual ~IRobotControl() = default;
+    };
+
+    template <typename T>
+    class RobotControl : public IRobotControl
+    {
+    public:
+        explicit RobotControl(ControlConfiguration controlConfiguration)
+            : m_controlConfiguration{std::move(controlConfiguration)} {}
+
+        void Activate(const AZ::Entity* entity) final
+        {
+            m_active = true;
+            if (!m_controlSubscription)
+            {
+                auto ros2Frame = entity->FindComponent<ROS2FrameComponent>();
+                AZStd::string namespacedTopic = ROS2Names::GetNamespacedName(
+                        ros2Frame->GetNamespace(),
+                        m_controlConfiguration.m_topic);
+
+                auto ros2Node = ROS2Interface::Get()->GetNode();
+                m_controlSubscription = ros2Node->create_subscription<T>(
+                    namespacedTopic.data(),
+                    m_controlConfiguration.m_qos.GetQoS(),
+                    std::bind(&RobotControl<T>::OnControlMessage, this, std::placeholders::_1));
```

Comment:
FYI: Using a lambda results in more [efficient code](https://clang.llvm.org/extra/clang-tidy/checks/modernize-avoid-bind.html) than using std::bind.

```suggestion
                   [this](const T& message) { OnControlMessage(message); });
```